model/task_history.go

```diff
@@ -394,17 +389,34 @@ func (self *taskHistoryIterator) GetDistinctTestNames(numCommits int) ([]string,
 			{"$unwind": fmt.Sprintf("$%v", testResultsKey)},
 			{"$group": bson.M{"_id": fmt.Sprintf("$%v.%v", testResultsKey, task.TestResultTestFileKey)}},
 		},
+		opts,
 	)
-
-	var output []bson.M
-
-	if err = pipeline.All(&output); err != nil {
-		return nil, errors.WithStack(err)
+	if err != nil {
+		return nil, errors.Wrapf(err, "error getting cursor")
+	}
+	if cursor == nil {
+		return nil, errors.New("nil cursor returned")
 	}
 
-	names := make([]string, 0)
-	for _, doc := range output {
-		names = append(names, doc["_id"].(string))
+	names := []string{}
+	for cursor.Next(ctx) {
+		select {
+		case <-ctx.Done():
+			_ = cursor.Close(ctx)
+			return nil, errors.New("context cancelled")
+		default:
+			res := bson.M{}
+			if err := cursor.Decode(&res); err != nil {
+				return nil, errors.Wrapf(err, "error decoding result")
+			}
+			// remove quotes from value
+			val := strings.ReplaceAll(res["_id"].(string), "\"", "")
```

Comment:
do you need to do this? I wouldn't think that quote in test names would cause problems

Reply:
Yeah, I added this because it was printing like: 
`"\"TestName\""`

Reply:
did you check if it was display weirdness with how you were testing it? If someone has quotes in their test names for whatever reason, we shouldn't be removing it

Reply:
With changes stashed, it's formatted correctly. With changes, it's formatted weirdly. I don't think people will have quotes in their testnames.